p2p/p2p.go

```diff
@@ -152,14 +136,27 @@ func (s *Service) receiveCoreMessages() {
 		// receive message from core service
 		msg := <-s.msgRec
 
-		log.Trace(
-			"Broadcasting message from core service",
-			"host", s.host.id(),
-			"type", msg.GetType(),
-		)
+		// check if non-status message
+		if msg.GetType() != StatusMsgType {
+
+			log.Trace(
+				"Broadcasting message from core service",
+				"host", s.host.id(),
+				"type", msg.GetType(),
+			)
 
-		// broadcast message to connected peers
-		s.host.broadcast(msg)
+			// broadcast message to connected peers
+			s.host.broadcast(msg)
+
+		} else {
+
+			// check if status exchange is enabled
+			if !s.host.noStatus {
+
+				// update host status message
+				s.status.setHostMessage(msg)
```

Comment:
so just to make sure, this saves the most current status message we get from the network, and then sends that if someone connects?

Reply:
correct, this saves the current status message for the host, which will be used when sending status messages to connected peers

Reply:
the host status message is generated in the core service and sent to the p2p service and then set (or updated) in the status submodule where it is used to validate status messages from peers

Reply:
also peer status messages are never sent to core service, just stored here as well when needing to generate `PeerInfo` for connected peers (`PeerInfo` is now accessible with `p2p.Peers()` which is needed for rpc and runtime)